include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -327,14 +304,16 @@ struct __parallel_scan_submitter<_CustomName, __internal::__optional_kernel_name
         auto __size_per_wg = __iters_per_witem * __wgroup_size;
         auto __n_groups = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __size_per_wg);
         // Storage for the results of scan for each workgroup
-        sycl::buffer<_Type> __wg_sums(__n_groups);
+
+        using _TempStorage = __result_and_scratch_storage<std::decay_t<_ExecutionPolicy>, _Type>;
+        _TempStorage __result_and_scratch{__exec, __n_groups + 1};
```

Comment:
The legacy algorithm here used to use the last element of the temp buffer as the return of the future.  This was convenient because it also happens to be the "final carry-out" or our desired return value. 

This PR changes the functionality when `__result_and_scratch_storage` is used to have an additional return value in addition to the  `__n_groups` tempoary wg sums, and for the scan patterns that use the return, we explicitly copy the last value of the temporary array into the explicit return value.  

`__result_and_scratch_storage` when handled by the future only has the option to include the value of the "return" in the future, unless we extend it to do something else.

With all that, I think the update `__ngroups + 1` is the appropriate value.